File: projects/tox4j/src/test/java/im/tox/tox4j/core/exceptions/ToxFriendDeleteExceptionTest.java
package im.tox.tox4j.core.exceptions;

import im.tox.tox4j.ToxCoreTestBase;
import im.tox.tox4j.core.ToxCore;
import org.junit.Test;

import static org.junit.Assert.assertArrayEquals;
import static org.junit.Assert.assertEquals;

public class ToxFriendDeleteExceptionTest extends ToxCoreTestBase {

  @Test
  public void testDeleteFriendTwice() throws Exception {
    try (ToxCore tox = newTox()) {
      addFriends(tox, 5);
      assertArrayEquals(tox.getFriendList(), new int[]{ 0, 1, 2, 3, 4 });
      tox.deleteFriend(2);
      assertArrayEquals(tox.getFriendList(), new int[]{ 0, 1, 3, 4 });
      try {
        tox.deleteFriend(2);
        fail();
      } catch (ToxFriendDeleteException e) {
        assertEquals(ToxFriendDeleteException.Code.FRIEND_NOT_FOUND, e.code());
      }
    }
  }

  @Test
  public void testDeleteNonExistentFriend() throws Exception {
    try (ToxCore tox = newTox()) {
      addFriends(tox, 5);
      assertArrayEquals(tox.getFriendList(), new int[]{ 0, 1, 2, 3, 4 });
      try {
        tox.deleteFriend(5);
        fail();
      } catch (ToxFriendDeleteException e) {
        assertEquals(ToxFriendDeleteException.Code.FRIEND_NOT_FOUND, e.code());
      }
    }
  }

}

File: projects/tox4j/src/test/java/im/tox/tox4j/core/exceptions/ToxFriendDeleteExceptionTest.scala
package im.tox.tox4j.core.exceptions

import im.tox.tox4j.testing.ToxTestMixin
import org.scalatest.FunSuite

final class ToxFriendDeleteExceptionTest extends FunSuite with ToxTestMixin {

  test("DeleteFriendTwice") {
    interceptWithTox(ToxFriendDeleteException.Code.FRIEND_NOT_FOUND) { tox =>
      addFriends(tox, 4)
      assert(tox.getFriendList sameElements Array(0, 1, 2, 3, 4))
      tox.deleteFriend(2)
      assert(tox.getFriendList sameElements Array(0, 1, 3, 4))
      tox.deleteFriend(2)
    }
  }

  test("DeleteNonExistentFriend") {
    interceptWithTox(ToxFriendDeleteException.Code.FRIEND_NOT_FOUND)(
      _.deleteFriend(1)
    )
  }

}
